tool/plugin/lib/runner.dart

```diff
@@ -93,13 +93,39 @@ testing=$testing
     // During instrumentation of FlutterProjectStep.form, which is a UTF-8 file.
     try {
       if (Platform.isWindows) {
+        if (spec.version == '4.0' || spec.version == '4.1') {
+          log('CANNOT BUILD ${spec.version} ON WINDOWS');
+          return 0;
+        }
         result = await exec('.\\gradlew.bat', command);
       } else {
-        result = await exec('./gradlew', command);
+        if (spec.version == '4.0' || spec.version == '4.1') {
+          return await runShellScript(command, spec);
+        } else {
+          result = await exec('./gradlew', command);
+        }
       }
     } finally {
       propertiesFile.writeAsStringSync(source);
     }
     return result;
   }
+
+  Future<int> runShellScript(List<String> command, BuildSpec spec) async {
```

Comment:
Could this have a more descriptive name? Like `gradleWithOldJava`?

Reply:
Sure, I can change it. This won't be around very long though.